### ISSUE TITLE
Consider pgr_dijkstra/pgr_ksp/pgr_drivingDistance and pgr_trsp(via mode) result seq value

### ISSUE THREAD

**Issue by sanak** [pgRouting/pgrouting#395]
In my understanding, result 1st column(`seq`) value is for sorting all rows,
so it should not be duplicated and started from not 1 but 0.

[pgr_dijkstra - Description of the return values](http://docs.pgrouting.org/v2.1.0-rc1/src/dijkstra/doc/dijkstra_v3.html#description-of-the-return-values) mentions as follows:
> seq:	INT relative position in the path. Has value 1 for the begining of the path.

But, if `seq` value is necessary for each path index, its naming should be another one (`idx` or something), and I still prefer that the value is started from 0.

**Comment by cvvergara** [comment]
Dijkstra 1 starting point 1 ending point:
from 2 to 3:
```
SELECT * FROM pgr_dijkstra(
                'SELECT id, source, target, cost, reverse_cost FROM edge_table',
                2, 3
        );
 seq | node | edge | cost | agg_cost
-----+------+------+------+----------
   1 |    2 |    4 |    1 |        0
   2 |    5 |    8 |    1 |        1
   3 |    6 |    9 |    1 |        2
   4 |    9 |   16 |    1 |        3
   5 |    4 |    3 |    1 |        4
   6 |    3 |   -1 |    0 |        5
(6 rows)
```
from 2 to 5:
```
SELECT * FROM pgr_dijkstra(
                'SELECT id, source, target, cost, reverse_cost FROM edge_table',
                2, 5
        );
 seq | node | edge | cost | agg_cost
-----+------+------+------+----------
   1 |    2 |    4 |    1 |        0
   2 |    5 |   -1 |    0 |        1
(2 rows)
```
```
from 2 to 3 and to 5:
the result is the union of the sets. its like making the above queries.
```
SELECT * FROM pgr_dijkstra(
                'SELECT id, source, target, cost, reverse_cost FROM edge_table',
                2, array[3,5]
        );
 seq | end_v | node | edge | cost | agg_cost
-----+-------+------+------+------+----------
   1 |     3 |    2 |    4 |    1 |        0
   2 |     3 |    5 |    8 |    1 |        1
   3 |     3 |    6 |    9 |    1 |        2
   4 |     3 |    9 |   16 |    1 |        3
   5 |     3 |    4 |    3 |    1 |        4
   6 |     3 |    3 |   -1 |    0 |        5
   1 |     5 |    2 |    4 |    1 |        0        <---  Note that a path begins here
   2 |     5 |    5 |   -1 |    0 |        1        <--- so node 5 its relative position in the path is 2
(8 rows)
```

**Comment by woodbri** [comment]
Why clutter the output when you can compute it directly like the following or like above using a SQL window function?

```
select end_v, sum(cost) as agg_cost from pgr_dijkstra(
                'SELECT id, source, target, cost, reverse_cost FROM edge_table',
                2, array[3,5]
        );

 end_v | agg_cost
-------+----------
     3 |        5
     5 |        1
( 2 rows)
```

**Comment by woodbri** [comment]
@sanak The sequence starts at 1 to be consistent with postgresql sequences that start from 1. All of the other commands that use sequence originally started with 1, although I have not taken inventory of them all lately. Also postgres is weird in that arrays by default are 1 based and not 0 based.

**Comment by sanak** [comment]
@woodbri 
Thanks for comment. 
But, pgRouting v2.0 used 0 as `seq` start value.
http://docs.pgrouting.org/2.0/en/src/dijkstra/doc/index.html#examples
So, I think that it should be mentioned in v2.1 NEWS and Changelog,
if it becomes a new standard.

**Comment by sanak** [comment]
By the way, pgr_trsp via mode result's `seq` values are also invalid.
http://docs.pgrouting.org/v2.1.0-rc1/src/trsp/doc/index.html#examples
And I think that it should be fixed before v2.1 release.
> An example query using vertex ids and via points:

<pre>
 seq | id1 | id2 | cost
-----+-----+-----+------
   1 |   1 |   1 |    1
   2 |   2 |   4 |    1
   3 |   7 |   8 |    1
   4 |   8 |   8 |    1
   5 |   7 |  10 |    1
   6 |  10 |  14 |    1
   7 |  13 |  14 |    1
   8 |  10 |  10 |    1
   9 |   7 |   7 |    1
  10 |   6 |   6 |    1
   4 |   5 |  -1 |    0
</pre>
* Last row `seq` value is skipped (starts from 4?).

> An example query using edge ids and vias:

<pre>
 seq | id1 | id2 | cost
-----+-----+-----+------
   0 |  -1 |   1 |  0.5
   1 |   2 |   4 |    1
   2 |   7 |   8 |    1
   3 |   8 |  11 |    1
   1 |  11 |  13 |    1
   2 |  12 |  15 |    1
   3 |   9 |   9 |    1
   4 |   8 |   8 |    1
   5 |   7 |   7 |    1
   6 |   6 |   6 |  0.5
(10 rows)
</pre>
* 1st path's `seq` value starts from 0, but 2nd path's `seq` value start from 1.

Also, duplicated `seq` value can't be used for sort order after joining the result to original edge table.
https://github.com/anitagraser/pgRoutingLayer/commit/1a8215a796033ccdd58def7f9bdf7164dab93342

**Comment by cvvergara** [comment]
For the V2.1 this was fixed:
http://docs.pgrouting.org/2.1/en/src/trsp/doc/index.html
(note the examples are generated with the current code)
Because the correct milestone is closed I am tagging this issue for 2.2.4
